crates/noirc_evaluator/src/ssa/node.rs

```diff
@@ -1228,3 +1245,24 @@ impl BinaryOp {
         )
     }
 }
+
+fn field_to_signed(f: FieldElement, n: u32) -> i128 {
+    assert!(n < 127);
+    let a = f.to_u128();
+    let pow_2 = 2_u128.pow(n);
+    if a < pow_2 {
+        a as i128
+    } else {
+        (a - 2 * pow_2) as i128
+    }
+}
+
+fn signed_to_field(a: i128, n: u32) -> FieldElement {
+    assert!(n < 126);
```

Comment:
Don't we allow `i126` along with `u126`?

Reply:
The assert is for the function to be correct, it is limited because of the way I use the i128, which is straightforward, so it is a quick win when n<126.

Reply:
Ah I see. Should we maybe then disallow `i126` earlier if it will eventually fail here? We are checking the bit size during evaluation here (https://github.com/noir-lang/noir/blob/d26bb40ce0f505090d13423d698af44df7ca0460/crates/noirc_evaluator/src/ssa/context.rs#L1117). We could adapt this to have a max of one less bit for signed integers.